math/exp.c

```diff
@@ -22,9 +22,13 @@
 double frexp(double x, int *exp)
 {
 	conv_t *conv = (conv_t *)&x;
-
 	*exp = 0;
 
+	if (isnan(x) != 0) {
+		errno = EDOM;
+		return NAN;
+	}
+
 	if ((x == 0.0) || (x == -0.0)) {
```

Comment:
There are checks for ±0 and NAN - what about ±INF?

Applies to other functions.